libs/utils/src/celix_convert_utils.c

```diff
@@ -107,25 +112,281 @@ long celix_utils_convertStringToLong(const char* val, long defaultValue, bool* c
+            *list = celix_arrayList_copy(defaultValue);
+            return *list ? status : CELIX_ENOMEM;
         }
+        return status;
+    }
+    return status;
+}
+
+celix_status_t celix_utils_addLongEntry(celix_array_list_t* list, const char* entry) {
+    bool converted;
+    long l = celix_utils_convertStringToLong(entry, 0L, &converted);
+    if (!converted) {
+        return CELIX_ILLEGAL_ARGUMENT;
+    }
+    return celix_arrayList_addLong(list, l);
+}
+
+celix_status_t celix_utils_convertStringToLongArrayList(const char* val,
+                                                        const celix_array_list_t* defaultValue,
+                                                        celix_array_list_t** list) {
+    return celix_utils_convertStringToArrayList(val, defaultValue, list, NULL, celix_utils_addLongEntry);
+}
+
+celix_status_t celix_utils_addDoubleEntry(celix_array_list_t* list, const char* entry) {
+    bool converted;
+    double d = celix_utils_convertStringToDouble(entry, 0.0, &converted);
+    if (!converted) {
+        return CELIX_ILLEGAL_ARGUMENT;
     }
-    if (converted) {
-        *converted = result != NULL;
+    return celix_arrayList_addDouble(list, d);
+}
+
+celix_status_t celix_utils_convertStringToDoubleArrayList(const char* val,
+                                                          const celix_array_list_t* defaultValue,
+                                                          celix_array_list_t** list) {
+    return celix_utils_convertStringToArrayList(val, defaultValue, list, NULL, celix_utils_addDoubleEntry);
+}
+
+celix_status_t celix_utils_addBoolEntry(celix_array_list_t* list, const char* entry) {
+    bool converted;
+    bool b = celix_utils_convertStringToBool(entry, true, &converted);
+    if (!converted) {
+        return CELIX_ILLEGAL_ARGUMENT;
     }
-    if (result == NULL && defaultValue != NULL) {
-        result = celix_version_copy(defaultValue);
+    return celix_arrayList_addBool(list, b);
+}
+
+celix_status_t celix_utils_convertStringToBoolArrayList(const char* val,
+                                                          const celix_array_list_t* defaultValue,
+                                                          celix_array_list_t** list) {
+    return celix_utils_convertStringToArrayList(val, defaultValue, list, NULL, celix_utils_addBoolEntry);
+}
+
+celix_status_t celix_utils_addStringEntry(celix_array_list_t* list, const char* entry) {
+    char* copy = celix_utils_strdup(entry);
+    if (!copy) {
+            return CELIX_ENOMEM;
     }
+    return celix_arrayList_add(list, copy);
+}
+
+celix_status_t celix_utils_convertStringToStringArrayList(const char* val,
+                                                          const celix_array_list_t* defaultValue,
+                                                          celix_array_list_t** list) {
+    return celix_utils_convertStringToArrayList(val, defaultValue, list, free, celix_utils_addStringEntry);
```

Comment:
If `defaultValue` is shallow copied into `list`, then `list` depends on `defaultValue`, i.e. `defaultValue` cannot be destroyed before `list`, which will be totally unexpected by our users.

The same remark applies to `celix_utils_convertStringToVersionArrayList`.